src/server/server.ts

```diff
@@ -1,127 +1,24 @@
-    return remoteExtensionHostAgentServer.handleServerError(err)
-  })
-
-  server.listen(options, async () => {
-    address = server.address()
-    if (address === null) {
-      throw new Error('Unexpected server address')
-    }
-
-    // eslint-disable-next-line no-console
-    console.log(`Server bound to ${typeof address === 'string' ? address : `${address.address}:${address.port} (${address.family})`}`)
-
-    await getRemoteExtensionHostAgentServer()
-  })
-
-  process.on('exit', () => {
-    server.close()
-    if (_remoteExtensionHostAgentServer != null) {
-      _remoteExtensionHostAgentServer.dispose()
-    }
-  })
+const thisWithVSCodeParams = globalThis as typeof globalThis & {
+  _VSCODE_PRODUCT_JSON: Partial<IProductService>
+  _VSCODE_PACKAGE_JSON: { version: string }
 }
 
-function parseRange (strRange: string) {
-  const match = strRange.match(/^(\d+)-(\d+)$/)
-  if (match != null) {
-    const start = parseInt(match[1]!, 10); const end = parseInt(match[2]!, 10)
-    if (start > 0 && start <= end && end <= 65535) {
-      return { start, end }
-    }
+// Initialize the product information for the server, including the extension gallery URL.
+thisWithVSCodeParams._VSCODE_PRODUCT_JSON = {
+  extensionsGallery: {
+    serviceUrl: 'https://open-vsx.org/vscode/gallery',
+    itemUrl: 'https://open-vsx.org/vscode/item',
+    resourceUrlTemplate: 'https://open-vsx.org/vscode/unpkg/{publisher}/{name}/{version}/{path}',
+    controlUrl: '',
+    nlsBaseUrl: '',
+    publisherUrl: ''
   }
-  return undefined
 }
-
-async function findFreePort (host: string, start: number, end: number) {
-  const testPort = (port: number) => {
-    return new Promise<boolean>((resolve) => {
-      const server = http.createServer()
-      server.listen(port, host, () => {
-        server.close()
-        resolve(true)
-      }).on('error', () => {
-        resolve(false)
-      })
-    })
-  }
-  for (let port = start; port <= end; port++) {
-    if (await testPort(port)) {
-      return port
-    }
-  }
-  return undefined
-}
-
-async function parsePort (host: string, strPort: string | undefined) {
-  if (strPort != null) {
-    let range
-    if (strPort.match(/^\d+$/) != null) {
-      return parseInt(strPort, 10)
-    } else if ((range = parseRange(strPort)) != null) {
-      const port = await findFreePort(host, range.start, range.end)
-      if (port !== undefined) {
-        return port
-      }
-      // Remote-SSH extension relies on this exact port error message, treat as an API
-      console.warn(`--port: Could not find free port in range: ${range.start} - ${range.end} (inclusive).`)
-      process.exit(1)
-    } else {
-      console.warn(`--port "${strPort}" is not a valid number or range. Ranges must be in the form 'from-to' with 'from' an integer larger than 0 and not larger than 'end'.`)
-      process.exit(1)
-    }
-  }
-  return 8000
+// @ts-ignore
+thisWithVSCodeParams._VSCODE_PACKAGE_JSON = {
+  version: '1.83.0'
```

Comment:
What about injecting it? like in the ProductService

Reply:
Oh interesting... I thought I had done that but maybe it got reverted or something.

Reply:
You need to add the `@rollup/plugin-replace` plugin in the rollup config for the server then

Reply:
or maybe instead of just setting the version, you can import the VSCode whole package.json file